### PR TITLE
Create multi-arch image

### PR DESCRIPTION
**Pull request by kelvie**
For linux arm64 and amd64, as per #586.

I moved the builder image into the same dockerfile, and bumped the Go
version on it. It didn't seem like the builder dockerfile worked with
the latest code anyway (the go modules require go 1.15 and higher).

This requires a recent enough docker install with buildx, as well as
an arm64 builder.

BASEIMAGE is changed to not specify an arch, so that the image will
build on its native arch in buildx.

Example image is on docker hub as:

    kelvie/node-problem-detector:v0.8.10-5-gb0fa610

### CODE REVIEW

**Comment by better0332** [comment]
I need this too

**Comment by yankay** [comment]
Greate PR.

**Comment by rotarur** [comment]
if works but the health-checker reports wrong kubelet status KubeletUnhealthy

**Comment by rotarur** [comment]
found the issue with kubelet. It uses http to get the status but without hostNetwork enabled it doesn't work. Now everything seems to work fine and in a cluster with mixed architectures, both amd64 and arm64 architectures, using multi-arch docker image

**Comment by acumino** [comment]
/assign @vteratipally
Can you please review this?

**Comment by otterley** [comment]
Hi maintainers! We have an increasing number of customers who would like to use NPD on arm64 CPUs. Is there anything I can do to help unblock this effort? Please feel free to contact me privately as well.

File: Makefile [hunk 1]
```diff
@@ -239,8 +240,8 @@ $(NPD_NAME_VERSION)-%.tar.gz: $(ALL_BINARIES) test/e2e-install.sh
 
 build-binaries: $(ALL_BINARIES)
 
-build-container: build-binaries Dockerfile
-	docker build -t $(IMAGE) --build-arg BASEIMAGE=$(BASEIMAGE) --build-arg LOGCOUNTER=$(LOGCOUNTER) .
+build-container: clean Dockerfile
```
Comment (otterley) [inline Makefile:243]: Needs

```
docker buildx create --use
```

to be run first. I recommend a prerequisite `.PHONY` target for this.

File: Makefile [hunk 2]
```diff
@@ -252,16 +253,20 @@ build-tar: $(TARBALL) $(ALL_TARBALLS)
...
 		-c 'cd /gopath/src/k8s.io/node-problem-detector/ && make build-binaries'
 
 push-container: build-container
+	# So we can push to docker hub by setting REGISTRY
+ifneq (,$(findstring gcr.io,$(REGISTRY)))
 	gcloud auth configure-docker
-	docker push $(IMAGE)
+endif
+	# Build should be cached from build-container
+	docker buildx build --push --platform $(DOCKER_PLATFORMS) -t $(IMAGE) --build-arg BASEIMAGE=$(BASEIMAGE) --build-arg LOGCOUNTER=$(LOGCOUNTER) .
```
Comment (otterley) [inline Makefile:269]: Needs

```
docker buildx create --use
```

to be run first. I recommend a prerequisite `.PHONY` target (e.g. `create-buildx-builder`) for this.

**Comment by otterley** [comment]
I don't think this PR is going to work as written, because it requires an arm64 build instance. The build nodes are all x86.

**Pull request merged**